account_move_group_restriction/security/account_move_group_restriction_security.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="group_restricted_account_management" model="res.groups">
+        <field name="name">Accounting / View Restricted Accounts</field>
+        <field name="category_id" ref="base.module_category_accounting_and_finance" />
+        <field
+            name="implied_ids"
+            eval="[Command.link(ref('account.group_account_user'))]"
+        />
+    </record>
+
+    <record id="account_move_line_account_group_rule" model="ir.rule">
+        <field name="name">Account Move Line by Account Group</field>
+        <field name="model_id" ref="account.model_account_move_line" />
+        <field name="domain_force">
+            ['|',
+                ('move_id.account_security_group_ids', '=', False),
+                ('move_id.account_security_group_ids', 'in', user.groups_id.ids)
```

Comment:
This allows users to access the move when they have ANY of the access groups involved (instead of all), which should not be something intended.

We may leave the design as is, since most use cases will involve only one such group, but we should add a ROADMAP.md to explain this limitation.